app/models/podcast.rb

```diff
@@ -40,7 +40,8 @@ def admins
   private
 
   def unique_slug_including_users_and_orgs
-    errors.add(:slug, "is taken.") if User.find_by(username: slug) || Organization.find_by(slug: slug) || Page.find_by(slug: slug)
+    slug_exists = User.exists?(username: slug) || Organization.exists?(slug: slug) || Page.exists?(slug: slug)
```

Comment:
here I replaced find_by which loads the entire object in memory, with exists? which only returns a boolean